service/awsconfig/framework.go

```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
 		}
 	}
 
-	// We provide a map of resource lists keyed by the version bundle version
-	// to the resource router.
-	versionedResources := map[string][]framework.Resource{
-		// Clusters without a version use the legacy resource.
-		"":      legacyResources,
-		"0.1.0": legacyResources,
-		"0.2.0": resourcesV2,
-		"1.0.0": legacyResources,
-		"2.0.0": resourcesV2,
-		"2.0.1": resourcesV2_0_1,
-		// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
-		"2.0.2": resourcesV2_0_1,
+	handlesFuncFunc := func(handledVersionBundles []string) func(obj interface{}) bool {
+		return func(obj interface{}) bool {
+			kvmConfig, err := key.ToCustomObject(obj)
+			if err != nil {
+				return false
+			}
+			versionBundleVersion := key.VersionBundleVersion(kvmConfig)
+
+			for _, v := range handledVersionBundles {
+				if versionBundleVersion == v {
+					return true
+				}
+			}
+
+			return false
+		}
+	}
+
+	var resourceSetV1 *framework.ResourceSet
```

Comment:
I named that v1 - this is the name of package where leagacy resource will spend the rest of its life.

Reply:
Very nice, I like that we keep legacy isolated in its final weeks!